src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,102 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int currentCapacity;
+    private int threshold;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        this.currentCapacity = DEFAULT_INITIAL_CAPACITY;
+        this.threshold = (int) (DEFAULT_INITIAL_CAPACITY * LOAD_FACTOR);
+        this.table = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshold) {
+            resize();
+        }
+        putValue(getBucketIndex(key), key, value);
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int hash = getBucketIndex(key);
+        Node<K, V> currentNode = table[hash];
+        while (currentNode != null) {
+            if (Objects.equals(key, currentNode.key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void transfer(Node<K, V>[] oldTable) {
+        for (Node<K, V> node : oldTable) {
+            Node<K,V> currentNode = node;
+            while (currentNode != null) {
+                putValue(getBucketIndex(currentNode.key), currentNode.key, currentNode.value);
+                currentNode = currentNode.next;
+            }
+        }
+    }
+
+    private void resize() {
+        if (size >= threshold) {
+            currentCapacity = currentCapacity * 2;
```

Comment:
```suggestion
            currentCapacity = currentCapacity * GROW_COEFFICIENT;
```
2 - it is a magic number